### ISSUE TITLE
unified-signatures incorrectly reports a violation and enters an infinite loop

### ISSUE THREAD

**Issue by kujon** [palantir/tslint#3204]
### Bug Report

- __TSLint version__: 5.4.2
- __TypeScript version__:2.5.0-dev.20170808
- __Running TSLint via__: CLI or Atom

#### Consider the following snippet:

```ts
// These overloads can be combined into one signature taking `T[][] | Array<T | T[]>`. (unified-signatures)
unnest<T>(x: Array<T | T[]>): T[];
```

So one attempts to fix it like requested:

```ts
These overloads can be combined into one signature taking `T[][] | T[][] | Array<T | T[]>`. (unified-signatures)
unnest<T>(x: T[][] | Array<T | T[]>): T[];
```

So one attempts to fix it like requested:

```ts
These overloads can be combined into one signature taking `T[][] | T[][] | T[][] | Array<T | T[]>`. (unified-signatures)
unnest<T>(x: T[][] | T[][] | Array<T | T[]>): T[];
```

So one attempts to fix it...

#### TSLint config
Just a regular `dtslint/dt.json`

#### Actual behavior
1. `unified-signatures` violation is reported
2. fix is made
3. `unified-signatures` violation is reported
4. fix is made
5. `unified-signatures` violation is reported
6. fix is made
7. `unified-signatures` violation...

#### Expected behavior
1. `unified-signatures` is not reported since `Array<T | T[]>` is already unified.
2. ...
3. profit

**Comment by ajafff** [comment]
I cannot reproduce this with the TSLint 5.7.0 and TypeScript 2.5.1

Can you please upgrade and verify that it's fixed? Otherwise I need a bit more information about your failing code.
